File: src/year2015/day02/part2.cc
#include <cstdint>
#include <string>

#include "../solutions.h"
#include "parse.h"

auto Year2015::Day02::Part2(const std::string& input) -> std::string {
  auto boxes = parse(input);
  uint32_t ribbonLength = 0;
  for (const auto& box : boxes) {
    ribbonLength += (2 * (box[0] + box[1]) + box[0] * box[1] * box[2]);
  }
  return std::to_string(ribbonLength);
}

File: src/year2015/day02/part2.rs
use super::parse::parse;
pub fn part2(input: String) -> Result<String, String> {
  let boxes = parse(input);
  if boxes.is_err() {
    return Err(boxes.unwrap_err());
  }
  let mut ribbon_length = 0;
  for curr_box in boxes.unwrap() {
    ribbon_length += 2 * (curr_box[0] + curr_box[1]) + curr_box[0] * curr_box[1] * curr_box[2];
  }
  return Ok(ribbon_length.to_string());
}
